From 511400b8ec0794588df88e2bb1c45cc6a4a51b1e Mon Sep 17 00:00:00 2001
From: Maximilian Bosch <maximilian@mbosch.me>
Date: Sun, 26 May 2019 23:10:02 +0200
Subject: [PATCH] Fix CMake configuration for Nix

* -fno-limit-debug-info: CLang specific, patch away to support both
  CLang and GCC.
* Dropped unnecessary `lvm-internal` and `daemonclient` references.
---
 CMakeLists.txt                | 1 -
 osquery/tables/CMakeLists.txt | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f545d6c9..cfe68ddb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -328,7 +328,6 @@ if (POSIX)
   # To be safe, only include them when building 'release' outputs.
   add_compile_options(
     "$<$<CONFIG:RELEASE>:-g>"
-    "$<$<CONFIG:RELEASE>:-fno-limit-debug-info>"
     "$<$<CONFIG:RELEASE>:-fPIE>"
     "$<$<CONFIG:RELEASE>:-fpie>"
     "$<$<CONFIG:RELEASE>:-fPIC>"
diff --git a/osquery/tables/CMakeLists.txt b/osquery/tables/CMakeLists.txt
index 470bdf1d..caa6ab6f 100644
--- a/osquery/tables/CMakeLists.txt
+++ b/osquery/tables/CMakeLists.txt
@@ -68,7 +68,7 @@ if(LINUX)
   set(TABLE_PLATFORM "linux")
 
   ADD_OSQUERY_LINK_ADDITIONAL("libresolv.so")
-  ADD_OSQUERY_LINK_ADDITIONAL("cryptsetup devmapper lvm2app lvm-internal daemonclient")
+  ADD_OSQUERY_LINK_ADDITIONAL("cryptsetup devmapper lvm2app")
   ADD_OSQUERY_LINK_ADDITIONAL("gcrypt gpg-error")
   ADD_OSQUERY_LINK_ADDITIONAL("blkid")
   ADD_OSQUERY_LINK_ADDITIONAL("ip4tc")
-- 
2.19.2

